src/editors/components/EditorFooter/index.test.jsx

```diff
@@ -1,104 +1,96 @@
 import React from 'react';
-import { render, screen } from '@testing-library/react';
-import userEvent from '@testing-library/user-event';
-import { mount } from 'enzyme';
-import EditorFooter from './EditorFooter';
-import EditorPageContext from './EditorPageContext';
-import { ActionStates } from './data/constants';
-import EditorPageProvider from './EditorPageProvider';
-import { saveBlock } from './data/api';
+import { shallow } from 'enzyme';
+import * as module from './index';
+import { selectors, thunkActions } from '../../data/redux';
+import { RequestKeys } from '../../data/constants/requests';
+import { saveTextBlock, navigateCallback } from '../../hooks';
 
-const locationTemp = window.location;
-beforeAll(() => {
-  delete window.location;
-  window.location = {
-    assign: jest.fn(),
-  };
-});
-afterAll(() => {
-  window.location = locationTemp;
-});
+jest.mock('../../data/redux', () => ({
+  thunkActions: {
+    app: {
+      saveBlock: jest.fn().mockName('thunkActions.app.saveBlock'),
+    },
+  },
+  selectors: {
+    app: {
+      isInitialized: jest.fn(state => ({ isInitialized: state })),
+      studioEndpointUrl: jest.fn(state => ({ studioEndpointUrl: state })),
+    },
+    requests: {
+      // eslint-disable-next-line no-unused-vars
+      isFailed: jest.fn((state, requestkey) => ({ isFailed: state })),
+    },
+  },
+}));
 
-jest.mock('./data/api', () => {
-  const originalModule = jest.requireActual('./data/api');
-  // Mock the default export and named export saveBlock
+jest.mock('.', () => {
```

Comment:
This... is not how I would necessarily have done this, but... is not wrong, and could arguably be cleaner for hook components.
However, should be more compact, and it is important to note that these methods should NOT just return a basic mock, but rather something inspectable, in order to verify the arguments.

```
jest.mock('.', () => ({
  // Require the original module to not be mocked...
  __esModule: true, // Use it when dealing with esModules
  ...jest.requireActual('./index'),
  handleCancelClicked: jest.fn(args => ({ handleCancelClicked: args })),
  handleSaveClicked: jest.fn(args => ({ handleSaveClicked: args })),
}));
```